packages/cms-cli/commands/init.js

```diff
@@ -29,45 +28,54 @@ const { addLoggerOptions, setLogLevel } = require('../lib/commonOpts');
 const { logDebugInfo } = require('../lib/debugInfo');
 
 const COMMAND_NAME = 'init';
+const TRACKING_STATUS = {
+  STARTED: 'started',
+  COMPLETE: 'complete',
+};
+const AUTH_METHOD_FLOW = {
+  [AUTH_METHODS.api.value]: {
+    prompt: async () => {
+      return promptUser(API_KEY_FLOW);
+    },
+    setup: async configData => {
+      createEmptyConfigFile({
+        deleteOnExitIfBlank: true,
+      });
+      updateDefaultPortal(configData.name);
+      updatePortalConfig(configData);
+    },
+  },
+  [AUTH_METHODS.oauth.value]: {
+    prompt: async () => {
+      return promptUser(OAUTH_FLOW);
+    },
+    setup: async configData => {
+      createEmptyConfigFile({
+        deleteOnExitIfBlank: true,
+      });
+      await authenticateWithOauth(configData);
+      process.exit();
+    },
+  },
+};
 
-const oauthConfigSetup = async ({ configPath }) => {
-  const configData = await promptUser(OAUTH_FLOW);
-
-  try {
-    createEmptyConfigFile();
-    process.on('exit', deleteEmptyConfigFile);
-    await authenticateWithOauth(configData);
-    process.exit();
-  } catch (err) {
-    logFileSystemErrorInstance(err, {
-      filepath: configPath,
-      configData,
-    });
-  }
-
-  trackCommandUsage(COMMAND_NAME, {
-    authType: AUTH_METHODS.oauth.value,
-  });
+const trackAuthMethodStatus = (authMethod, status) => {
+  return trackCommandUsage(`${authMethod} ${status}`);
```

Comment:
Since what we're tracking here is not usage of a command but progress within a flow initiated by a command, I think we should create a separate tracking helper with a different `action` property.